tests/test_modeling_common.py

```diff
@@ -1539,6 +1552,49 @@ def check_same_values(layer_1, layer_2):
             # self.assertTrue(model.transformer.wte.weight.shape, model.lm_head.weight.shape)
             # self.assertTrue(check_same_values(model.transformer.wte, model.lm_head))
 
+    def test_tied_model_weights_key_ignore(self):
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+        for model_class in self.all_model_classes:
+            model_tied = model_class(config)
+            with tempfile.TemporaryDirectory() as d:
+                model_tied.save_pretrained(d)
+
+                # We are nuking ALL weights on file, so every parameter should
+                # yell on load. We're going to detect if we yell too much, or too little.
+                with open(os.path.join(d, "pytorch_model.bin"), "wb") as f:
+                    torch.save({}, f)
+                model_reloaded, infos = model_class.from_pretrained(d, output_loading_info=True)
+
+                prefix = f"{model_reloaded.base_model_prefix}."
+                params = dict(model_reloaded.named_parameters())
+                params.update(dict(model_reloaded.named_buffers()))
+                # param_names = set(k[len(prefix) :] if k.startswith(prefix) else k for k in params.keys())
+                param_names = set(k[len(prefix) :] if k.startswith(prefix) else k for k in params.keys())
+
+                missing_keys = set(infos["missing_keys"])
+
+                extra_missing = missing_keys - param_names
+                # missed_missing = param_names - missing_keys
+
+                if extra_missing:
+                    import ipdb
+
+                    ipdb.set_trace()
+
+                self.assertEqual(
+                    extra_missing,
+                    set(),
+                    f"This model {model_class.__name__} might be missing some `keys_to_ignore`: {extra_missing} when"
+                    " tying its weights",
+                )
+
+                # self.assertEqual(
```

Comment:
I deactivated this part because `token_type_ids` and `position_ids` would yell pretty much everywhere.

I have no idea of the perfect solution for this.
I **think** `position_ids` should be a non-parameter if it's not, or a parameter if it is learned or something. 
Given how many models it affect we could add an escape hatch for just these peculiar names.

But I think flagging **actually** too inclusive `_keys_to_ignore` can provide good value, especially since we're using regexp so it's easy to make a mistake, or modify the model later in such a way that it doesn't reflect properly anymore in those keys.